### ISSUE TITLE
Widget Inspector shows too many widgets

### ISSUE THREAD

**Issue by johnpryan** [flutter/devtools#6841]
When I open the Widget Inspector for the counter app, I get a simplified version of the widget tree, but when I open it for DartPad  ([pkgs/sketch_pad](https://github.com/dart-lang/dart-pad/tree/main/pkgs/sketch_pad)), I see a very complex widget tree, with widgets like View and MediaQuery, that aren't part of my application code.

**Counter app:**

![Screenshot 2023-11-28 at 10 01 05 AM](https://github.com/flutter/devtools/assets/1145719/887ebad1-150f-404b-83c2-17c6ebd752af)

**DartPad:**

![Screenshot 2023-11-28 at 10 05 04 AM](https://github.com/flutter/devtools/assets/1145719/88728157-12d9-4be6-81ce-961713d45078)

**Comment by kenzieschmoll** [comment]
Can you check your Package Directories in the Inspector settings and post a screenshot of what is there?

**Comment by kenzieschmoll** [comment]
Delete the flutter SDK from the list of package directories. I'm not sure how this was added to the list. @CoderDake could there be a bug in our package directory detection logic?

**Comment by kenzieschmoll** [comment]
> @jacob314 and I were just discussing something similar where an internal customer's widget tree is comprised of primarily Provider widgets, making it quite user-unfriendly.

Were these Provider widgets ones that they defined in their app? Or were they widgets from the internals of package:provider?

**Comment by johnpryan** [comment]
> Delete the flutter SDK from the list of package directories. I'm not sure how this was added to the list. @CoderDake could there be a bug in our package directory detection logic?

I'm not sure either, but that fixed it for me.

> @jacob314 and I were just discussing something similar where an internal customer's widget tree is comprised of primarily Provider widgets, making it quite user-unfriendly.

I'm seeing the correct number of Provider widgets I would expect in DartPad, so I'm not sure if this is related or not.

**Comment by elliette** [comment]
Summarizing what @jacob314 just told me in the office:

In `inspector_service.dart`, we make the assumption that the first widget in your widget tree is owned by your application. However, @goderbauer recently landed a change to add a `View` widget as the first widget in the tree. (I'm seeing the `View` widget both in DartPad's widget tree, and in the widget tree of an internal customer that is incorrectly getting the flutter package added to their project directories.) We will need to update our detection logic to non assume that the first widget in the tree is owned by the application itself.

@goderbauer did the `View` widget changes make it into this latest Flutter stable release? Thanks!

**Comment by goderbauer** [comment]
> did the View widget changes make it into this latest Flutter stable release? Thanks!

Yes, the change that adds the `View` widget to the root of every Flutter app made it into the stable release.

> We will need to update our detection logic to non assume that the first widget in the tree is owned by the application itself.

Because of that change this assumption is now incorrect. I wasn't aware that DevTools made this assumption...

**Comment by jacob314** [comment]
Odd that
https://github.com/flutter/devtools/blob/master/packages/devtools_app/test/inspector/inspector_service_test.dart#L96
didn't catch this issue. We should make sure those tests are still running correctly.

Good news is @elliette is working on  a fix that will stop relying on the top widget in the tree being user created.
Once the fix is in it should be cherry-picked as this makes the inspector useless for CLI users.
